routes/songlist.js

```diff
@@ -203,55 +200,65 @@ router.route('/:song_id/add-translation')
-                lang: lang
-            }, function(err, translation) {
-                if (err) {
-                    res.status(400).send('error ' + err)
-                }
-                var newSong = new Song({
-                    title: req.body.title_t,
-                    author: song.author,
-                    year: song.year,
-                    lang: lang,
-                    contributor: req.user.username,
-                    copyright: req.body.copyright_t,
-                    lyric: stringArr_t.slice(0),
+        req.checkBody('title_t', 'Title is required').notEmpty()
+        req.checkBody('lyric_t', 'Lyric is required').notEmpty()
+        var messages = req.validationErrors()
+
+        if (messages) {
+            res.render('addTranslation', {
+                messages: messages
+            })
+        } else {
+            var lang = req.body.lang_t
+            var stringArr_t = req.body.lyric_t.split(/\r?\n|\//)
+            Song.findOne({
                     source: song.id,
-                    oriSong: song.title,
-                    timeAdded: Date.now()
-                })
-                if (translation) {
-                    newSong.v = translation.v + 1
-                } else {
-                    newSong.v = 1;
-                }
-                newSong.save(function(err) {
+                    lang: lang
+                }, function(err, translation) {
                     if (err) {
-                        res.status(400).send('error saving new song ' + err)
+                        res.status(400).send('error ' + err)
+                    }
+                    var newSong = new Song({
+                        title: req.body.title_t,
+                        author: song.author,
+                        year: song.year,
+                        lang: lang,
+                        contributor: req.user.username,
+                        copyright: req.body.copyright_t,
+                        lyric: stringArr_t.slice(0),
+                        source: song.id,
+                        oriSong: song.title,
+                        timeAdded: Date.now()
+                    })
+                    if (translation) {
+                        newSong.v = translation.v + 1
                     } else {
-                        res.redirect('/songlist/' + song_id)
+                        newSong.v = 1;
                     }
+                    newSong.save(function(err) {
+                        if (err) {
+                            res.status(400).send('error saving new song ' + err)
+                        } else {
+                            res.redirect('/songlist/' + song_id)
+                        }
+                    })
                 })
-            })
-            .sort({
-                _id: -1
-            }).limit(1)
+                .sort({
+                    _id: -1
+                }).limit(1)
+        }
+        // if (stringArr_t.length != song.lyric.length) {
```

Comment:
we might need this if you want to prevent the user to add a song with different number of lines on the lyric dont we?